src/utils/encoding.ts

```diff
@@ -325,25 +330,29 @@ export function unwrapRawExchangeRate(option: Option<UnsignedFixedPoint>): Unsig
     return option.isSome ? (option.value as UnsignedFixedPoint) : undefined;
 }
 
-export async function encodeVaultId(assetRegistry: AssetRegistryAPI, id: InterbtcPrimitivesVaultId): Promise<string> {
+export async function encodeVaultId(assetRegistry: AssetRegistryAPI, loansAPI: LoansAPI, id: InterbtcPrimitivesVaultId): Promise<string> {
     const [wrappedCurrency, collateralCurrency] = await Promise.all([
-        currencyIdToMonetaryCurrency(assetRegistry, id.currencies.wrapped),
-        currencyIdToMonetaryCurrency(assetRegistry, id.currencies.collateral),
+        currencyIdToMonetaryCurrency(assetRegistry, loansAPI, id.currencies.wrapped),
+        currencyIdToMonetaryCurrency(assetRegistry, loansAPI, id.currencies.collateral),
     ]);
-    const wrappedId = isForeignAsset(wrappedCurrency) ? wrappedCurrency.id.toString() : wrappedCurrency.ticker;
+    const wrappedId = isForeignAsset(wrappedCurrency) ? wrappedCurrency.foreignAsset.id.toString() :
```

Comment:
technically a wrapped asset can be neither `foreign` nor `lendtoken` but this check doesn't harm

Reply:
Ha, pretty sure that was me adding that a while back. What can I say? 
I'm just _that_ forward thinking prepping vaults-for-everyone ;)